images/velero/tests/docker-test.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+
+install_velero(){
+  velero install --namespace velero \
+               --no-default-backup-location \
+               --use-volume-snapshots=false \
+               --no-secret \
+               --image ${IMAGE_REGISTRY}/${IMAGE_REPOSITORY}:${IMAGE_TAG}
+}
+
+wait_for_pod() {
+  local label=$1
+  local namespace=$2
+  local timeout=$3
+  local start_time=$(date +%s)
+  local end_time=$((start_time + timeout))
+
+  while true; do
+    if [ "$(date +%s)" -gt "$end_time" ]; then
+      echo "Timeout waiting for pod with label $label to be created"
+      exit 1
+    fi
+
+    if kubectl get pods -n "$namespace" -l "$label" 2>/dev/null | grep -q Running; then
+      break
+    fi
+
+    sleep 5
+  done
+}
+
+test_velero(){
+
+  wait_for_pod "component=velero" "velero" 300
+
+  VELERO_POD_NAME=$(kubectl get pods -n velero -l component=velero -o jsonpath='{.items[0].metadata.name}')
+  STATUS=$(kubectl get pod $VELERO_POD_NAME -n velero -o jsonpath='{.status.phase}')
+
+  if [ "$STATUS" == "Running" ]; then
+    echo "Velero pod is running"
+  else
+    echo "Velero pod is not running"
+  fi
+
+  # Create a test backup
+  velero backup create my-backup --include-namespaces default
+
+  # Check if the backup is created
+  if ! velero backup get | grep -q "my-backup"; then
```

Comment:
I think I ran into an issue in my local test where this failed, but it didn't throw an error when running via terraform. I'm trying to reproduce just to be sure.

Reply:
yeah, I see this for the backup:

```
bash-5.2# velero backup get
NAME        STATUS             ERRORS   WARNINGS   CREATED   EXPIRES   STORAGE LOCATION   SELECTOR
my-backup   FailedValidation   0        0          <nil>     29d       default            <none>
```

Reply:
This is what I see now, so I think there needs to be a flag set somewhere during backup creation is all:

```
bash-5.2# velero describe backup my-backup
Name:         my-backup
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/resource-timeout=10m0s
              velero.io/source-cluster-k8s-gitversion=v1.29.2+k3s1
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=29

Phase:  FailedValidation

Validation errors:  an existing backup storage location wasn't specified at backup creation time and the server default 'default' doesn't exist. Please address this issue (see `velero backup-location -h` for options) and create a new backup. Error: BackupStorageLocation.velero.io "default" not found
```

From there, I think the `velero backup get` command should also check the status is successful, right now the `grep -q` will always return true whether the backup succeeds or not.